app/src/util/itertools.ts

```diff
@@ -101,13 +109,13 @@ function* filter<T>(predicate: (x: T) => boolean, input: Generator<T>): Generato
 }
 
 function* dropwhile<T>(predicate: (x: T) => boolean, input: Generator<T>): Generator<T> {
+  let dropping = true;
   for (const item of input) {
-    if (!predicate(item)) {
+    dropping = dropping && predicate(item);
+    if (!dropping) {
```

Comment:
Nice catch, I did not know that generators are not reusable (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of#do_not_reuse_generators). Please also update `takewhile`